data/campaigns/Eastern_Invasion/utils/ai.cfg

```diff
@@ -0,0 +1,345 @@
+    {MODIFY_AI_DELETE_GOAL {SIDE} "*"}
+    {MODIFY_SIDE_AI ({SIDE}) ([avoid]
+    x,y=0,0
+[/avoid])}
+
+{MODIFY_AI_DELETE_ASPECT {SIDE} grouping "*"}
+{MODIFY_AI_DELETE_ASPECT {SIDE} aggression "*"}
+{MODIFY_AI_DELETE_ASPECT {SIDE} caution "*"}
+{MODIFY_AI_DELETE_ASPECT {SIDE} leader_value "*"}
+
+{MODIFY_SIDE_AI ({SIDE}) (grouping={GROUPING})}
+{MODIFY_SIDE_AI ({SIDE}) (aggression={AGGRESSION})}
+{MODIFY_SIDE_AI ({SIDE}) (caution={CAUTION})}
+{MODIFY_SIDE_AI ({SIDE}) (leader_value=0)}
+#enddef
+
+#define VARY_AI_BY_SCHEDULE SIDE
+    {VARY_AI_BY_SCHEDULE_ENEMY {SIDE} 1}
+#enddef
+
+#define VARY_AI_BY_SCHEDULE_ENEMY SIDE ENEMY
+    {VARY_AI_BY_SCHEDULE_ENEMY_TIME_OF_DAY {SIDE} {ENEMY} morning afternoon}
+#enddef
+
+#define VARY_AI_BY_SCHEDULE_ENEMY_NIGHTTIME SIDE ENEMY
+    {VARY_AI_BY_SCHEDULE_ENEMY_TIME_OF_DAY {SIDE} {ENEMY} first_watch second_watch}
+#enddef
+
+#define VARY_AI_BY_SCHEDULE_WINTER SIDE
+    {VARY_AI_BY_SCHEDULE_ENEMY_TIME_OF_DAY {SIDE} 1 midday midday}
+#enddef
+
+#define AND_NO_ENEMIES_NEAR_MY_LEADER SIDE
+    [not]
+        [have_unit]
+            side={SIDE}
+            canrecruit=yes
+            [filter_adjacent]
+                is_enemy=yes
+            [/filter_adjacent]
+        [/have_unit]
+    [/not]
+#enddef
+
+#define VARY_AI_BY_SCHEDULE_ENEMY_TIME_OF_DAY SIDE ENEMY TIME1 TIME2
+    [store_time_of_day]
+    [/store_time_of_day]
+    [if]
+        [variable]
+            name=time_of_day.id
+            equals={TIME1}
+        [/variable]
+
+        {AND_NO_ENEMIES_NEAR_MY_LEADER {SIDE}}
+        [then]
+            {MODIFY_SIDE_AI ({SIDE}) ({GOAL_AVOID_SIDE_UNLESS_DEFENDING_LEADER 9 {SIDE} {ENEMY} 3})} # avoid staying too close
+            {MODIFY_SIDE_AI ({SIDE}) ({GOAL_SEEK_SIDE                          8        {ENEMY} 9})} # but seek staying close enough
+
+            {MODIFY_SIDE_AI ({SIDE}) (aggression=-9)}
+            {MODIFY_SIDE_AI ({SIDE}) (caution=9)}
+            {MODIFY_SIDE_AI ({SIDE}) (grouping=defensive)}
+        [/then]
+    [/if]
+
+    [store_time_of_day]
+    [/store_time_of_day]
+    [if]
+        [variable]
+            name=time_of_day.id
+            equals={TIME2}
+        [/variable]
+
+        {AND_NO_ENEMIES_NEAR_MY_LEADER {SIDE}}
+        [then]
+            {MODIFY_SIDE_AI ({SIDE}) ({GOAL_AVOID_SIDE_UNLESS_DEFENDING_LEADER 9 {SIDE} {ENEMY} 1})} # avoid staying too close
+            {MODIFY_SIDE_AI ({SIDE}) ({GOAL_SEEK_SIDE                          8        {ENEMY} 7})} # but seek staying close enough
+
+            {MODIFY_SIDE_AI ({SIDE}) (aggression=-9)}
+            {MODIFY_SIDE_AI ({SIDE}) (caution=9)}
+            {MODIFY_SIDE_AI ({SIDE}) (grouping=defensive)}
+        [/then]
+    [/if]
+
+#enddef
+
+#define RETREAT_WHEN_WEAK SIDE COUNT GOALS
+    [if]
+        [have_unit]
+            side={SIDE}
+            count={COUNT}
+        [/have_unit]
+
+        {AND_NO_ENEMIES_NEAR_MY_LEADER {SIDE}}
+
+        [lua]
+            code=<< return wml.variables.turn_number>1 >>
```

Comment:
As before, this should either be a `[variable]` tag or use `wesnoth.current.turn`.